requests/adapters.py

```diff
@@ -344,13 +348,20 @@ def request_url(self, request, proxies):
         scheme = urlparse(request.url).scheme
 
         is_proxied_http_request = (proxy and scheme != 'https')
+        is_using_https_forwarding = (
+            proxies_kwargs and proxies_kwargs.get('use_forwarding_for_https', False)
+        )
+
+        is_using_https_proxy = False
         using_socks_proxy = False
         if proxy:
             proxy_scheme = urlparse(proxy).scheme.lower()
             using_socks_proxy = proxy_scheme.startswith('socks')
+            is_using_https_proxy = proxy_scheme.startswith('https')
 
         url = request.path_url
-        if is_proxied_http_request and not using_socks_proxy:
+        if (is_proxied_http_request and not using_socks_proxy or
+                (is_using_https_proxy and is_using_https_forwarding)):
```

Comment:
nit; this conditional is getting a kind of hard to parse at first glance. Perhaps we could define a couple variables like `is_forwarded_https_proxy` and `is_http_socks_proxy` to simplify the concepts.

e.g.
```python
if is_http_socks_proxy or is_forwarded_https_proxy:
```